src/docdb/tree/DocDBAccountTreeItemBase.ts

```diff
@@ -24,22 +24,38 @@ export abstract class DocDBAccountTreeItemBase extends DocDBTreeItemBase<Databas
     public readonly childTypeLabel: string = "Database";
 
 
-    constructor(parent: AzExtParentTreeItem, id: string, label: string, endpoint: string, masterKey: string, isEmulator: boolean | undefined, readonly databaseAccount?: DatabaseAccountGetResults) {
+    constructor(
+        parent: AzExtParentTreeItem,
+        id: string,
+        label: string,
+        endpoint: string,
+        credentials: CosmosDBCredential[],
+        isEmulator: boolean | undefined,
+        readonly databaseAccount?: DatabaseAccountGetResults
+    ) {
         super(parent);
         this.id = id;
         this.label = label;
         this.root = {
             endpoint,
-            masterKey,
+            credentials,
             isEmulator,
-            getCosmosClient: () => getCosmosClient(endpoint, masterKey, isEmulator)
+            getCosmosClient: () => getCosmosClient(endpoint, credentials, isEmulator)
         };
 
-        this.valuesToMask.push(id, endpoint, masterKey);
+        const keys = credentials
+            .map((cred) => cred.type === "key" ? cred.key : undefined)
+            .filter((value): value is string => value !== undefined);
+        this.valuesToMask.push(id, endpoint, ...keys);
     }
 
     public get connectionString(): string {
-        return `AccountEndpoint=${this.root.endpoint};AccountKey=${this.root.masterKey}`;
+        const firstKey = this.root.credentials.filter((cred): cred is CosmosDBKeyCredential => cred.type === "key")[0];
```

Comment:
🧐

Reply:
The 'lib' in tsconfig is too old to support .at. Changing it results in multiple build error to surface. I'll skip using .at in this PR.

Reply:
Good catch btw.